packages/core/src/nodes.rs

```diff
@@ -378,17 +378,39 @@ pub struct Template {
 
 impl std::hash::Hash for Template {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-        std::ptr::hash(self.roots as *const _, state);
-        std::ptr::hash(self.node_paths as *const _, state);
-        std::ptr::hash(self.attr_paths as *const _, state);
+        // If debug assertions are enabled, rust is probably built with a lower
+        // optimization level which will not merge identical statics. If it is,
+        // then we hash by contents
+        if cfg!(debug_assertions) {
```

Comment:
I think `debug_assertions` is the closest we can get to checking if that optimization is likely active, but it can fail in some cases like if you disable `debug_assertions` in debug mode for faster build times

Reply:
This should be more reliable now. We can check if function pointers are merged in a way that hopefully gets resolved at compile time. If they are merged, this is a build that is optimized enough to merge statics